File: Projects/rosalind/IPRB.java
import java.io.*;
import java.util.*;

public class IPRB {
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(System.in);
		int k = in.nextInt(), m = in.nextInt(), n = in.nextInt();
		int all = (k + m + n) * (k + m + n - 1) / 2;
		double pmm = (double) m * (m-1) / 8 / all;
		double pmn = (double) m * n / 2 / all;
		double pnn = (double) n * (n-1) / 2 / all;
		System.out.println(1 - pmm - pmn - pnn);
		in.close();
	}
}

File: Projects/rosalind/IPRB.kt
package rosalind

import java.io.IOException
import java.util.*

object IPRB {
    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        val k = stdin.nextInt()
        val m = stdin.nextInt()
        val n = stdin.nextInt()
        val all = (k + m + n) * (k + m + n - 1) / 2
        val pmm = m.toDouble() * (m - 1) / 8 / all
        val pmn = m.toDouble() * n / 2 / all
        val pnn = n.toDouble() * (n - 1) / 2 / all
        println(1 - pmm - pmn - pnn)
        stdin.close()
    }
}